src/cpp/rtps/common/PayloadInfo_t.hpp

```diff
@@ -65,14 +65,14 @@ struct PayloadInfo_t
         assert(change.serializedPayload.data == nullptr);
         assert(change.serializedPayload.length == 0);
 
-        change.payload_owner(payload_owner);
-        payload_owner = nullptr;
-
         change.serializedPayload = payload;
         payload.data = nullptr;
         payload.length = 0;
         payload.pos = 0;
         payload.max_size = 0;
+
```

Comment:
I think we could get rid of `PayloadInfo_t` and use `SerializedPayload_t` directly.